packages/prosemirror-lwdita/src/request.ts

```diff
@@ -109,17 +120,56 @@ export function processRequest(): undefined | { key: string, value: string }[] {
 
     try {
       const parameters = getAndValidateParameterValues(currentUrl);
+      
       if (typeof parameters === 'string') {
+        if(parameters === 'invalidParams') {
+          //TODO(YB): Redirect to referer if it exists
+          //TODO(YB): Redirect to Petal error page if referer doesn't exist
+        }
         showNotification(parameters);
-      }
 
-      if (typeof parameters === 'object') {
+        return undefined;
+      } else if (typeof parameters === 'object' && !parameters.some(param => isOAuthCodeParam(param.key))) {
         // Store the parameters in localStorage for reading the values after the OAuth flow
         // TODO: After a successful GitHub Authentication, read the user parameters from localStorage and clear it afterwards
         localStorage.setItem('userParams', JSON.stringify(parameters));
 
         // Redirect to GitHub OAuth page
         redirectToGitHubOAuth();
+      } else if (typeof parameters === 'object' && parameters.some(param => isOAuthCodeParam(param.key))) {
+        //TODO(YB): These Params should passed with the OAuth redirect URL
+        // get the stored parameters from localStorage
+        const storedParams = localStorage.getItem('userParams');
+        if(!storedParams) {
+          // TODO: Redirect to Petal error page
+          // we should never reach here
+          return undefined;
+        }
+        // in case of an error, the user did not authenticate the app
+        const errorParam = parameters.find(param => param.key === 'error');
+        if(errorParam) {
+          //TODO(YB): redirect to petal error page
+          // Petal error page should have a button to redirect to the referer
+          return undefined;
+        }
+
+        // exchange the code for an access token
+        const codeParam = parameters.find(param => param.key === 'code');
+        if (!codeParam) return undefined; // I don't understand why this is necessary as the previous if statement should have caught this
+        const code = codeParam.value;
+        
+
+        exchangeOAuthCodeForAccessToken(code).then(token => {
+          localStorage.setItem('token', token);
+        }).catch(e => {
+          console.error(e);
```

Comment:
@marmoure After a successful authorization and redirect to the Petal application, this error is produced: `request.ts:165 SyntaxError: Unexpected token '<', "<!doctype "... is not valid JSON`

Reply:
@marmoure This fixed it: https://github.com/evolvedbinary/prosemirror-lwdita/pull/391/commits/c488fc085b6774c2349587a1e4f8f627c26d55fb